backend.native/tests/runtime/workers/worker10.kt

```diff
@@ -118,4 +119,28 @@ val semaphore = AtomicInt(0)
     semaphore.increment()
     future.result
     worker.requestTermination().result
+}
+
+fun <T: Any> ensureWeakIs(weak: WeakReference<T>, expected: T?) {
+    assertEquals(expected, weak.get())
+}
+
+val stableHolder = StableRef.create(("hello" to "world").freeze())
+
+@Test fun runTest4() {
+    val worker = Worker.start()
+    semaphore.value = 0
+    val future = worker.execute(TransferMode.SAFE, { WeakReference(stableHolder.get()) }) {
+        ensureWeakIs(it, "hello" to "world")
```

Comment:
This test doesn't reproduce the original issue.
If this worker task is modified to be like that in previous tests, then the issue gets reproduced.